docs/user_guide/selection/ProbeFeatureSelection.rst

```diff
@@ -0,0 +1,224 @@
+        confirm_variables=False
+    )
+
+    sel.fit(X_train, y_train)
+
+With :code:`fit()`, the transformer:
+
+    - creates `n_probes` number of probe features using provided distribution(s)
+    - uses cross-validation to fit the provided estimator
+    - calculates the feature importance score for each variable, including probe features
+    - if there are multiple probe features, calculate the average importance score
+    - identifies features to drop because their importance scores are less than that of the probe feature(s)
+
+
+In the attribute :code:`probe_features`, we find the pseudo-randomly generated variable(s):
+
+.. code:: python
+
+    sel.probe_features_.head()
+
+           gaussian_probe_0
+    0         -0.694150
+    1          1.171840
+    2          1.074892
+    3          1.698733
+    4          0.498702
+
+
+The attribute :code:`feature_importances_` shows each variable's feature importance:
+
+.. code:: python
+
+    sel.feature_importances.head()
+
+    mean radius        0.058463
+    mean texture       0.011953
+    mean perimeter     0.069516
+    mean area          0.050947
+    mean smoothness    0.004974
+
+In the attribute :code:`features_to_drop_`, we find the variables that were not selected:
+
+.. code:: python
+
+    sel.features_to_drop_
+
+    ['mean symmetry',
+     'mean fractal dimension',
+     'texture error',
+     'smoothness error',
+     'concave points error',
+     'fractal dimension error']
+
+We see that the :code:`features_to_drop` have feature importance scores that are less
+than the probe feature's score:
+
+.. code:: python
+
+    vars_to_drop = sel.features_to_drop_
+    vars_to_display = vars_to_drop + ["gaussian_probe_0"]
+
+    mean symmetry              0.003698
+    mean fractal dimension     0.003455
+    texture error              0.003595
+    smoothness error           0.003333
+    concave points error       0.003548
+    fractal dimension error    0.003576
+    gaussian_probe_0           0.003783
+
+With :code:`transform()`, we can go ahead and drop the six features with feature importance score
+less than `gaussian_probe_0` variable:
+
+.. code:: python
+
+    Xtr = sel.transform(X_test)
+
+    Xtr.shape
+
+    (114, 24)
+
+
+And, finally, we can also obtain the names of the features in the final transformed dataset:
+
+.. code:: python
+
+    sel.get_feature_names_out()
+
+    ['mean radius',
+     'mean texture',
+     'mean perimeter',
+     'mean area',
+     'mean smoothness',
+     'mean compactness',
+     'mean concavity',
+     'mean concave points',
+     'radius error',
+     'perimeter error',
+     'area error',
+     'compactness error',
+     'concavity error',
+     'symmetry error',
+     'worst radius',
+     'worst texture',
+     'worst perimeter',
+     'worst area',
+     'worst smoothness',
+     'worst compactness',
+     'worst concavity',
+     'worst concave points',
+     'worst symmetry',
+     'worst fractal dimension']
+
```

Comment:
@Morgan-Sell this document is just great! Thank you so much.

If I get to be greedy, I could I ask you to add a few more snippets of code to create 3 probe features with the param `"all"` and then plot histograms of those features to show the users their distribution?

Reply:
How can I say no to profesora che? ;)